taskcluster/xpi_taskgraph/worker_types.py

```diff
@@ -53,15 +53,58 @@ def build_scriptworker_signing_payload(config, task, task_def):
     )
 
 
+@payload_builder("shipit-shipped", schema={Required("release-name"): text_type})
+def build_push_apk_payload(config, task, task_def):
+    worker = task["worker"]
+
+    task_def["payload"] = {"release_name": worker["release-name"]}
+
+
+# NOTE: copied scriptworker-github from fenix w/few modifications
 @payload_builder(
-    "shipit-shipped",
+    "scriptworker-github",
     schema={
+        Required("upstream-artifacts"): [
+            {
+                Required("taskId"): taskref_or_string,
+                Required("taskType"): text_type,
+                Required("paths"): [text_type],
+            }
+        ],
+        Required("artifact-map"): [object],
+        Required("action"): text_type,
+        Required("git-tag"): text_type,
+        Required("git-revision"): text_type,
+        Required("github-project"): text_type,
+        Required("is-prerelease"): bool,
         Required("release-name"): text_type,
     },
 )
-def build_push_apk_payload(config, task, task_def):
+def build_github_release_payload(config, task, task_def):
     worker = task["worker"]
 
+    task_def["tags"]["worker-implementation"] = "scriptworker"
+
+    owner, repo_name = worker["github-project"].split("/")
+    if owner not in ["mozilla-releng", "mozilla-extensions"]:
+        # TODO: throw an error here
+        pass
```

Comment:
We'll probably have some scopes issues otherwise. I think I'm ok with that; we probably don't need this check.